### PR TITLE
Zypper.find_products returns an empty array when no products are found / Card 2783

### PR DESCRIPTION
**Pull request by lagartoflojo**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
@kalabiyau is your reviewer :dancers: check https://trello.com/c/XFa3iRkP/2783-2-cnt-do-not-throw-exception-when-product-is-not-found 
@kalabiyau: Please review this pull request using our guidelines: 
* test for acceptance criteria / functionality 
* check if the new code is covered with tests 
* check for unintended consequences 
* encourage usage of the boyscout rule 
* make sure the code is architected in the best way 
* check that no unnecessary technical debt got introduced 
* make sure that no unnecessary FIXMEs or TODOs got added

**Comment by lagartoflojo** [comment]
@kalabiyau I'm not happy that we still log an error (`log.error`). When we call zypper with a product that doesn't exist, zypper has a non-zero exit status, so our `execute` method logs the error. I don't know if this is desirable.

**Comment by lagartoflojo** [comment]
@kalabiyau Also, Trello card says

>*please use bnc in the changelog so it will pass submission*
> https://bugzilla.suse.com/show_bug.cgi?id=943451

What does that mean?

File: lib/suse/connect/zypper.rb [hunk 1]
```diff
@@ -84,6 +84,8 @@ def find_products(identifier)
           zypper_out = call("--xmlout --non-interactive search -s -t product #{Shellwords.escape(identifier)}", false)
           xml_doc = REXML::Document.new(zypper_out, compress_whitespace: [])
           xml_doc.elements.to_a('stream/search-result/solvable-list/solvable').map(&:to_hash)
+        rescue
```
Comment (kalabiyau) [inline lib/suse/connect/zypper.rb]: maybe we can do here `rescue Connect::ZypperError` to be able to be more specific

**Comment by lagartoflojo** [comment]
@kalabiyau could you check it now please?

**Comment by kalabiyau** [comment]
LGTM

**Pull request merged**